var/spack/repos/builtin/packages/py-tensorflow/package.py

```diff
@@ -0,0 +1,692 @@
+
+            # Do you want to use clang as CUDA compiler?
+            env.set('TF_CUDA_CLANG', '0')
+
+            # Please specify which gcc nvcc should use as the host compiler
+            env.set('GCC_HOST_COMPILER_PATH', spack_cc)
+
+            cuda_paths = [
+                spec['cuda'].prefix,
+                spec['cudnn'].prefix,
+            ]
+
+            # Do you wish to build TensorFlow with TensorRT support?
+            if '+tensorrt' in spec:
+                env.set('TF_NEED_TENSORRT', '1')
+
+                cuda_paths.append(spec['tensorrt'].prefix)
+
+                # Please specify the TensorRT version you want to use
+                env.set('TF_TENSORRT_VERSION',
+                        spec['tensorrt'].version.up_to(1))
+
+                # Please specify the location where TensorRT is installed
+                env.set('TENSORRT_INSTALL_PATH', spec['tensorrt'].prefix)
+            else:
+                env.set('TF_NEED_TENSORRT', '0')
+                env.unset('TF_TENSORRT_VERSION')
+
+            # Please specify the CUDA SDK version you want to use
+            env.set('TF_CUDA_VERSION', spec['cuda'].version.up_to(2))
+
+            # Please specify the cuDNN version you want to use
+            env.set('TF_CUDNN_VERSION', spec['cudnn'].version.up_to(1))
+
+            if '+nccl' in spec:
+                cuda_paths.append(spec['nccl'].prefix)
+
+                # Please specify the locally installed NCCL version to use
+                env.set('TF_NCCL_VERSION', spec['nccl'].version.up_to(1))
+
+                # Please specify the location where NCCL is installed
+                env.set('NCCL_INSTALL_PATH', spec['nccl'].prefix)
+                env.set('NCCL_HDR_PATH', spec['nccl'].prefix.include)
+            else:
+                env.unset('TF_NCCL_VERSION')
+
+            # Please specify the comma-separated list of base paths to
+            # look for CUDA libraries and headers
+            env.set('TF_CUDA_PATHS', ','.join(cuda_paths))
+
+            # Please specify the location where CUDA toolkit is installed
+            env.set('CUDA_TOOLKIT_PATH', spec['cuda'].prefix)
+
+            # Please specify the location where CUDNN library is installed
+            env.set('CUDNN_INSTALL_PATH', spec['cudnn'].prefix)
+
+            # Please specify a list of comma-separated CUDA compute
+            # capabilities you want to build with. You can find the compute
+            # capability of your device at:
+            # https://developer.nvidia.com/cuda-gpus.
+            # Please note that each additional compute capability significantly
+            # increases your build time and binary size, and that TensorFlow
+            # only supports compute capabilities >= 3.5
+            if spec.variants['cuda_arch'].value != 'none':
+                capabilities = ','.join('{0:.1f}'.format(
+                    float(i) / 10.0) for i in spec.variants['cuda_arch'].value)
```

Comment:
@Sinan81 I think this string formatting will be much more stable so that you don't get weird things like `'6.499999999'` instead of `'6.5'`.

Reply:
What should we put if `cuda_arch=none`? I'm trying to build on a local cluster. CUDA is installed, but I don't see any NVIDIA GPU on the cluster. I want to test anyway, but I don't know what my CUDA capabilities are.